packages/keiko-ui/src/app/components/desktop/hooks/workspace-persistence.ts

```
@@ -778,51 +850,117 @@ export function parsePersistedWindows(raw: string | null): AppWindow[] | null {
+  conn: Connection,
+  aliases: ReadonlyMap<string, string>,
+  windowIds: ReadonlySet<string>,
+): string | undefined {
+  if (typeof conn.boundChatWindowId !== "string" || conn.boundChatWindowId.length === 0) {
+    return undefined;
+  }
+  const remapped = remapWindowId(conn.boundChatWindowId, aliases);
+  return windowIds.has(remapped) ? remapped : undefined;
+}
+
+function sanitizeConnection(
```

ESLint

[error] Function 'sanitizeConnection' has a complexity of 17. Maximum allowed is 10.

(complexity)